platform/windows/display_server_windows.cpp

```diff
@@ -3473,6 +3480,14 @@ LRESULT DisplayServerWindows::WndProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARA
 					min_max_info->ptMaxTrackSize.x = windows[window_id].max_size.x + decor.x;
 					min_max_info->ptMaxTrackSize.y = windows[window_id].max_size.y + decor.y;
 				}
+				if (windows[window_id].borderless) {
+					Rect2i screen_rect = screen_get_usable_rect(window_get_current_screen(window_id));
+					int border = ((screen_get_size(window_get_current_screen(window_id)) == screen_rect.size) ? 1 : 0);
```

Comment:
What's that supposed to do? If you are trying to make window bigger to hide border in non-exclusive fullscreen, it will not work on many GPUs (depends on a driver, some account only for client area) and will break mulitwindow apps like Godot editor.

Reply:
Here actually, I am trying to add the border instead (when the taskbar and other panels are hidden), this border is not needed when the taskbar is present already (is it needed? 😅).
The border was previously being added withing the if block at line 1870.

Reply:
This code seems problematic.. right?
Making the window smaller shouldn't be the right way.
should we undo this? we will have to disable maximization support (through the menu) in borderless mode, if we decide to undo this.
Or is there a better solution to allow maximization in borderless mode when the taskbar is hidden?

Reply:
Changing window size is not a good option, that's why `WS_BORDER` was applied to maximized window (to keep window size and reduce client area), but I'm not sure if it would work with `WS_MAXIMIZE`.

Reply:
I would always disable Alt+Space for borderless and full-screen if someone is using borderless window, it's not expected to be maximizable (and if it is, a custom decorations and Alt+Space menu should be implemented).

Reply:
`WS_BORDER` does work with `WS_MAXIMIZE` just fine (I tried just now). But the problem then becomes how do we update the style when maximization is done through the menu. Because our `update_style` is not called (I think) when maximization happens through the menu.

The common part that runs in both cases (through code and through menu) is here.

Reply:
The problematic border has been removed, now we use `WS_BORDER` if the maximized window covers the entire screen👍

Reply:
Hi 😊
> if someone is using borderless window, it's not expected to be maximizable

I think it is expected to be borderless maximizable, as on linux, godot does support borderless maximization, and window menu also works with borderless windows.
For fullscreen, the menu has been disabled ✅